helix-core/src/main/java/org/apache/helix/model/IdealState.java

```diff
@@ -721,9 +719,7 @@ private RebalanceMode normalizeRebalanceMode(IdealStateModeProperty mode) {
       String rebalancerName = getRebalancerClassName();
       if (rebalancerName != null) {
         if (rebalancerName.equals(JobRebalancer.class.getName())
-            || rebalancerName.equals(WorkflowRebalancer.class.getName())
-            || rebalancerName.equals(GenericTaskRebalancer.class.getName())
-            || rebalancerName.equals(FixedTargetTaskRebalancer.class.getName())) {
+            || rebalancerName.equals(WorkflowRebalancer.class.getName())) {
```

Comment:
Would this break backward compatibility for existing long-running TF resources?

Reply:
I think we are safe because the remove ones have been deprecated around 4 years ago. Perhaps it is time to do some cleaning for TF code base.